network/p2p/readConnection.go

```diff
@@ -44,7 +45,6 @@ func newReadConnection(ctx context.Context,
 	}
 
 	c := readConnection{
-		ctx:        ctx,
```

Comment:
what was this for?

Reply:
Sorry, I think I need to add this back

Reply:
this was context that the middleware passed in so that when the middleware was stopping it could tell the readConnection to stop and exit as well.

Reply:
line [72](https://github.com/onflow/flow-go/pull/2079/files#diff-f8c0a7795966e3b2bf497901c5128a0e462624582e4a6a40d133f7041300c148L72) has an `rc.ctx` while would need this to be set.

Reply:
I think removing this will cause a panic here:
https://github.com/onflow/flow-go/blob/b4f37e0c537af6c437147604ebabd43e6265f1c3/network/p2p/readConnection.go#L72

Maybe we could remove it from `readConnection` completely and pass the context into `receiveLoop`

Reply:
@jordanschalm that's a much better idea